util/tr_openstudio.rb

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/ruby
+
+# Simple OSMeasure Test Runner
+# 2016.10.06 RPG/NREL
+# TODO add a scanner(s) for Rspec et al. test defs
+
+require 'optparse'
+require 'csv'
+require 'json'
+
+# options parser
+
+options = {}
+
+optparse = OptionParser.new do|opts|
+
+  opts.banner = "\nUsage: #{$0} [options]\n\nScans all specified test files for (Minitest) tests and \
+  \nruns them; errors reported to file."
+   
+  opts.separator ""
+  opts.separator "Options:"
+  
+  options[:wd] = Dir.pwd
+  opts.on( '-d', '--directory <value>', "Working directory (default: '.')") do|f|
+    options[:wd] = f
+  end
+  
+  options[:test_file] = nil
+  opts.on( '-t', '--test <file1,file2...>', "Test filename(s)" ) do|f|
+    options[:test_file] = f.split(",")
+  end
+
+  options[:tests_map] = nil
+  opts.on( '-m', '--mask <regex>', "Regex search for test file(s)" ) do|f|
+    options[:tests_map] = f
+  end
+
+  options[:envs] = ["lib"]
+  opts.on( '-e', '--env <path1,path2,...>', "Ruby environment path(s) (default: 'lib')" ) do|f|
+    options[:envs] = f.split(",")
+  end
+  
+  opts.on( '-f', '--fetch', "Fetch measures from BCL and exit" ) do|f|
+    options[:bcl_fetch] = f
+  end
+
+  options[:bcl_query] = 'NREL'
+  opts.on( '-b', '--bcl <query>', "BCL measure query (default: 'NREL')" ) do|f|
+    options[:bcl_query] = f
+  end
+
+  options[:test_bcl] = false
+  opts.on( '-a', '--auto', "Test NREL BCL measures" ) do|f|
+    options[:test_bcl] = f
+  end
+
+  opts.on_tail("-h", "--help", "Show this message") do
+    puts opts
+    exit
+  end
+
+end.parse!
+
+
+# Fetch measures from BCL
+if options[:bcl_fetch] || options[:test_bcl]
+	puts "INFO: Fetching measures tagged '#{options[:bcl_query]}' from BCL..."
+	require 'bcl'
+	require 'bcl/version'
+	bcl = BCL::ComponentMethods.new
+	bcl.login   # sets the BCL URL
+	bcl.measure_metadata(options[:bcl_query])
+  puts 'BCL fetch complete.'
+	exit if options[:bcl_fetch]
+end
+
+tests = []
+measures = []
+if options[:test_bcl]
+  measures = Dir.glob("measures/parsed/**")
+  tests = Dir.glob("measures/parsed/**/tests/*_[tT]est.rb")
+
+elsif !options[:test_file] && !options[:tests_map]
+	puts "#{$0}: ERROR: No test(s) specified, use '-t --test', '-m --mask', or try '#{$0} -h' for help."
+	exit
+
+# TODO support both filespecs
+elsif 
+  options[:test_file] && !options[:tests_map]
+	tests = Dir.glob(options[:test_file])
```

Comment:
If the user types -t measures/foo it will, yeah. But the user can type whatever they wish. This is the problem with the measure xml update loop. It ASSumes the measures are in ./measures...